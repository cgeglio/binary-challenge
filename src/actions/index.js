export const login = user => ({
  type: 'LOGIN',
  user
})

export const logout = user => ({
  type: 'LOGOUT',
  user
})

export const addCards = cards => ({
  type: 'ADD_CARDS',
  cards
})

export const removeCards = cards => ({
  type: 'REMOVE_CARDS',
  cards
})

export const addQuestion = question => ({
  type: 'ADD_QUESTION',
  question
})

export const removeQuestion = question => ({
  type: 'REMOVE_QUESTION',
  question
})

export const addFortune = fortune => ({
  type: 'ADD_FORTUNE',
  fortune
})

export const removeFortune = fortune => ({
  type: 'REMOVE_FORTUNE',
  fortune
})

export const addReading = currentReading => ({
  type: 'ADD_READING',
  currentReading
})

export const removeReading = currentReading => ({
  type: 'REMOVE_READING',
  currentReading
})

export const addFavorite = favorite => ({
  type: 'ADD_FAVORITE',
  favorite
})

export const removeFavorite = favorite => ({
  type: 'REMOVE_FAVORITE',
  favorite
})

export const resetFavorites = favorites => ({
  type: 'RESET_FAVORITES',
  favorites
})
