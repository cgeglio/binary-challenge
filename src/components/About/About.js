import React from 'react';
import { Link } from 'react-router-dom';
import './About.scss';

const About = () => {
  return (
    <section className='about'>
      <h2>About Golden City Tarot</h2>
      <p>Golden City Tarot started in 1924 with a deck of cards and a dream. We are the premier destination for fortune telling in the Pacific Northwest. Tarot not your thing? We have crystal balls, palm reading, and aura reading, too. Remember, the future is golden at Golden City Tarot!</p>
      <h3>Visit Us</h3>
      <p>620 Treat Avenue</p>
      <p>San Francisco, CA 94110</p>
      <p>Phone: (415) 555-5555</p>
      <Link to='/home'><button className='back-btn'>Home</button></Link>
    </section>
  )
}

export default About;
