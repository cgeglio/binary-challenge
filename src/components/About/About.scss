@import url('https://fonts.googleapis.com/css?family=Croissant+One&display=swap');
@import "../../mixins.scss";

.about {
  @include flex();
}

.about p {
  font-size: 1.2em;
  margin: 7px;
  text-align: center;
  width: 50%;
}

.about h2 {
 font-family: 'Croissant One';
 font-size: 2em;
 margin-bottom: 20px;
}

.about h3 {
  font-size: 1.5em;
  margin: 30px 0px 0px 12px;
}
