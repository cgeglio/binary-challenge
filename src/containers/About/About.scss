@import "../../variables.scss";
@import "../../mixins.scss";

.about {
  @include flex();
}

.about p {
  font-size: 1.2em;
  margin: 7px;
  text-align: center;
  width: 50%;
}

.about h2 {
 font-family: $primary-font;
 font-size: 2em;
 margin-bottom: 20px;
}

.about h3 {
  font-size: 1.5em;
  margin: 30px 0px 0px 12px;
}
