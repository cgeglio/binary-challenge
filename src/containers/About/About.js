import React from 'react';
import { Link } from 'react-router-dom';
import './About.scss';
import { removeQuestion } from '../../actions';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export const About = (props) => {
  return (
    <section className='about fade-in'>
      <h2>About Golden City Tarot</h2>
      <p>Golden City Tarot started in 1924 with a deck of cards and a dream. We are the premier destination for fortune telling in the Pacific Northwest. Tarot not your thing? We have crystal balls, palm reading, and aura reading, too. Remember, the future is golden at Golden City Tarot!</p>
      <h3>Visit Us</h3>
      <p>620 Treat Avenue</p>
      <p>San Francisco, CA 94110</p>
      <p>Phone: (415) 555-5555</p>
      <Link to='/home'><button onClick={() => props.resetQuestion('question')}className='back-btn'>Home</button></Link>
    </section>
  )
}

export const mapDispatchToProps = dispatch => ({
  resetQuestion: question => (dispatch(removeQuestion(question)))
})

export default connect(null, mapDispatchToProps)(About);

About.propTypes = {
  resetQuestion: PropTypes.func
}
