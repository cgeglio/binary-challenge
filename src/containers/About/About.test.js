import React from 'react';
import { About, mapDispatchToProps } from './About';
import { removeQuestion } from '../../actions';
import { shallow } from 'enzyme';

describe('About', () => {

  let mockResetQuestion;
  let wrapper;

  beforeEach(() => {
    mockResetQuestion = jest.fn().mockImplementation();
    wrapper = shallow(<About resetQuestion={mockResetQuestion} />);
  });

  describe('About container/component', () => {
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should call resetQuestion when the home button is clicked', () => {
      wrapper.find('.back-btn').simulate('click');
      expect(mockResetQuestion).toHaveBeenCalledWith('question');
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the removeQuestion action when resetQuestion is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = removeQuestion('question');
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.resetQuestion('question');
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
