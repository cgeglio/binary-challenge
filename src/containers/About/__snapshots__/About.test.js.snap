// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About About container/component should match the snapshot 1`] = `
<section
  className="about fade-in"
>
  <h2>
    About Golden City Tarot
  </h2>
  <p>
    Golden City Tarot started in 1924 with a deck of cards and a dream. We are the premier destination for fortune telling in the Pacific Northwest. Tarot not your thing? We have crystal balls, palm reading, and aura reading, too. Remember, the future is golden at Golden City Tarot!
  </p>
  <h3>
    Visit Us
  </h3>
  <p>
    620 Treat Avenue
  </p>
  <p>
    San Francisco, CA 94110
  </p>
  <p>
    Phone: (415) 555-5555
  </p>
  <Link
    to="/home"
  >
    <button
      className="back-btn"
      onClick={[Function]}
    >
      Home
    </button>
  </Link>
</section>
`;
